pkg/helm/helm.go

```diff
@@ -87,14 +87,11 @@ func (h *Hydrator) templateArgs(ctx context.Context, destDir string) ([]string,
 }
 
 func (h *Hydrator) appendValuesArgs(args []string) ([]string, error) {
-	var values map[string]interface{}
-	err := json.Unmarshal([]byte(h.Values), &values)
-	if err != nil {
-		return []string{}, fmt.Errorf("failed to unmarshal helm.values, error: %w", err)
-	}
-	for key, val := range values {
-		args = append(args, "--set", fmt.Sprintf("%s=%v", key, val))
+	valuesPath := "tmp/chart-values.yaml"
+	if err := ioutil.WriteFile(valuesPath, []byte(h.Values), 0644); err != nil {
+		return []string{}, fmt.Errorf("failed to create values file: %w", err)
 	}
+	args = append(args, "--values", valuesPath)
```

Comment:
https://github.com/GoogleContainerTools/kpt-config-sync/blob/16fc6bafc040b5de3bb11110c495dc91bf0cddbf/cmd/helm-sync/main.go#L40 needs to be updated to reflect this change.

Reply:
Done